packages/agreement-process/src/routers/AgreementRouter.ts

```diff
@@ -592,6 +593,36 @@ const agreementRouter = (
         return res.status(errorRes.status).send(errorRes);
       }
     })
+    .post(
+      "/internal/agreement/:agreementId/signedContract",
+      async (req, res) => {
+        const ctx = fromAppContext(req.ctx);
+
+        try {
+          validateAuthorization(ctx, [INTERNAL_ROLE]);
+
+          const { agreementId } = req.params;
+          const agreementContract = AgreementSignedContract.parse(req.body);
+
+          const { metadata } =
+            await agreementService.internalAddAgreementSignedContract(
+              unsafeBrandId(agreementId),
+              agreementContract,
+              ctx
+            );
+          setMetadataVersionHeader(res, metadata);
```

Comment:
Is there a need for the set of the metadata version header? 
We usually put it for routes called by apiv2 that needs polling of the result.